### DISCUSSION TITLE
Licensing of Coraza ports to other languages

### DISCUSSION THREAD

**Discussion by SassonNir** [corazawaf/coraza#820] [Q&A] [answered]
Hello,
Recently I ported the core of Coraza into Python (and at the time of writing working on the seclang implementation).
The port is more then just bindings, I went through all the necessary files and translated the Go code to Python and also all of the tests and currently they all pass to the latest commit (at the time of writing).
I mostly kept the original structure with minor changes to adapt for Python importing system.
Since this mostly started as a personal project meant for others to use as baselines for their own Python ports I tried to mostly retain the comments of the code so it would be easier to cross reference to the original library 

Since I want to publish the code GitHub and I'm kinda new to open source publishing I wanted to know under what licensing I can publish it under, can I publish it under MIT? or do I need to keep the original license Apache 2?

I want to make sure I'm doing everything ok so there wont be any issues down the line.
Thanks in advance for any help

**Comment by jptosso** [comment, marked as answer]
Hey! Glad you are enjoying the project.

I'm not a lawyer but I think OSI compatible licenses are coraza-friendly. Apache, MIT, and 3-BSD.
You can also combine coraza with licenses like GPL, but you must make sure to state your changes and keep the license.

https://www.apache.org/legal/resolved.html

**Reply by SassonNir** [reply]
Thank you for the quick response, duly noted, and I'll update the discussion when it goes live.

**Comment by jcchavezs** [comment]
I wonder if before you go live with a port of something that took to years
to build and a year to polish and make it prod performance we could make a
case using https://github.com/wasmerio/wasmer-python so you can consume
coraza instead of rewriting it.

Not saying you can't make it, instead I am saying we can join forces that would benefit both communities.
